lightning/src/chain/channelmonitor.rs

```diff
@@ -1129,7 +1129,7 @@ impl<Signer: Sign> ChannelMonitor<Signer> {
 		&self,
 		updates: &ChannelMonitorUpdate,
 		broadcaster: &B,
-		fee_estimator: &F,
+		fee_estimator: &LowerBoundedFeeEstimator<F>,
```

Comment:
Seems we run into a few tricky issues here as we cannot clone a Deref to a FeeEstimator so that we can wrap it further down the line. Unless I'm missing something subtle and rusty. The main problem is we take in `&F` here and in some other places and we can't move out of the shared reference when we need to wrap.

Reply:
Alright, let me go that direction and see :)

Reply:
Seems to do the trick!